### DISCUSSION TITLE
Why does the setup run Taiga after installation only when in silent mode?

### DISCUSSION THREAD

**Discussion by sorashi** [erengy/taiga#1123] [Questions] [answered]
https://github.com/erengy/taiga/blob/bd2be25b62461c7eb49d8403f2b65e03e73d7dfb/setup/Taiga.nsi#L236-L240C12

Is this a mistake? Shouldn't it be configured like `SaveInstDir` above it? `IfSilent 0 +2`.

Currently, the setup runs the application only if the setup is run in silent mode with the `/S` flag.

I don't know whether this is intended behavior, so I'm not making an issue yet. Maybe you have a good reason for it :)

I'm asking because of this issue in a package I maintain, for reference: https://github.com/sorashi/chocolatey-packages/issues/2

**Comment by soredake** [comment]
I think i understand now why is this happening. Taiga is meant to be always-running application, so starting it after installation is good thing, but as i understand installer does not check if taiga is running before installation or is it installed so it starts it in fresh silent installation too.

**Comment by erengy** [comment, marked as answer]
Silent mode is used by Taiga's updater here:
https://github.com/erengy/taiga/blob/b6f440e48b2827dd9b0c680861cf6c498748c306/src/taiga/update.cpp#L263-L266

The process works as following:

- User agrees to upgrade Taiga to its latest version.
- Taiga downloads the installer and executes it using `/S` and `/D` flags.
- Installer ~sends a `WM_DESTROY` message to Taiga and~ waits 3 seconds for Taiga to close.
    *(Correction: In silent mode, installer skips sending the message.)*
- Installer silently extracts the files and restarts Taiga for a seamless upgrade.

This has been the case since 35342e54e907e7f47d9dc6056aaa0126fbffa953, for about ten years now. I didn't think of any other use case for the `/S` flag at the time. In retrospect, I should have probably used an additional parameter to restart the application.
